src/group_theory/sylow.lean

```diff
@@ -432,4 +440,35 @@ begin
   rwa [h, card_bot] at key,
 end
 
+
+instance subsingleton_of_normal {p : ℕ} [fact p.prime] [fintype (sylow p G)] (P : sylow p G)
+  (h : (P : subgroup G).normal) : subsingleton (sylow p G) :=
+begin
+  apply subsingleton.intro,
+  intros Q R,
+  obtain ⟨x, h1⟩ := @exists_smul_eq G (sylow p G) _ _ P Q,
+  rw sylow.smul_eq_of_normal at h1,
+  obtain ⟨x, h2⟩ := @exists_smul_eq G (sylow p G) _ _ P R,
+  rw sylow.smul_eq_of_normal at h2,
+  simp only [← h1, ← h2],
+end
+
+section pointwise
+
+open_locale pointwise
+
+instance characteristic_of_normal {p : ℕ} [fact p.prime] [fintype (sylow p G)] (P : sylow p G)
+  (h : (P : subgroup G).normal) :
+  (P : subgroup G).characteristic :=
+begin
+  apply characteristic_iff_map_eq.mpr,
+  intros Φ,
+  change (Φ • P).to_subgroup = P.to_subgroup,
+  suffices h : Φ • P = P, by {congr, exact h},
+  haveI := sylow.subsingleton_of_normal P h,
+  apply subsingleton.elim,
```

Comment:
I haven't open this up in vscode, but if you use `exact subsingleton.elim (fill in) (these arguments)`, then you should be able to avoid the `haveI`.

Reply:
Not with an `@` in front of `subsingleton.elim`, I think, which is why I avoided that?